### PR TITLE
pybind11: add 2.5.0 + cache cmake + test_package fixes

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **pybind11/2.5.0**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/pybind11/all/test_package/conanfile.py [hunk 1]
```diff
@@ -9,10 +9,17 @@ class TestPackageConan(ConanFile):
...
     def build(self):
         cmake = CMake(self)
+        cmake.definitions["Python_ADDITIONAL_VERSIONS"] = "{}.{}".format(sys.version_info.major,sys.version_info.minor)
         cmake.configure()
         cmake.build()
 
+    @property
+    def _python_available(self):
+        # FIXME: this can be removed once a python interpreter is available in CCI
+        return getattr(sys, "frozen", False)
```
Comment (madebr) [inline recipes/pybind11/all/test_package/conanfile.py:19]:
```suggestion
        return getattr(sys, "frozen", True)
```

File: recipes/pybind11/all/test_package/CMakeLists.txt [hunk 2]
```diff
@@ -1,8 +1,6 @@
 cmake_minimum_required(VERSION 2.8.11)
 project(test_package CXX)
 
-set(CMAKE_VERBOSE_MAKEFILE TRUE)
```
Comment (Croydon) [inline recipes/pybind11/all/test_package/CMakeLists.txt:4]: You remove this from every recipe you editing

We all should probably agree if we use that or not for stringency. Pinging some people for opinions // @danimtb @SSE4 @uilianries @ericLemanissier

Comment (SSE4) [reply recipes/pybind11/all/test_package/CMakeLists.txt:4]: we have discussed about it in the past, but I can't find the conversation (so many notifications, you know). I'll create new issue in the hooks repo.

Comment (SSE4) [reply recipes/pybind11/all/test_package/CMakeLists.txt:4]: https://github.com/conan-io/hooks/issues/188

File: recipes/pybind11/all/conanfile.py [hunk 3]
```diff
@@ -16,17 +16,25 @@ class PyBind11Conan(ConanFile):
...
 
+    _cmake = None
+
     def source(self):
         tools.get(**self.conan_data["sources"][self.version])
         os.rename("{}-{}".format(self.name, self.version), self._source_subfolder)
 
+    def requirements(self):
+        # self.requires("cpython/x.y/z") or self.requires("pypy/x.y/z")
+        self.output.info("This recipe requires a python SDK which is not available (yet) on CCI")
```
Comment (SSE4) [inline recipes/pybind11/all/conanfile.py:27]: there are few libraries, include this one and boost, capable of producing python modules written in C++. they usually need python development headers and libraries. we need sort of conan package to abstract it away, similar to [python-dev-config](https://github.com/bincrafters/conan-python_dev_config).

File: recipes/pybind11/all/conanfile.py [hunk 4]
```diff
@@ -16,17 +16,25 @@ class PyBind11Conan(ConanFile):
...
     _source_subfolder = "source_subfolder"
 
+    _cmake = None
+
     def source(self):
         tools.get(**self.conan_data["sources"][self.version])
         os.rename("{}-{}".format(self.name, self.version), self._source_subfolder)
 
+    def requirements(self):
+        # self.requires("cpython/x.y/z") or self.requires("pypy/x.y/z")
```
Comment (uilianries) [inline recipes/pybind11/all/conanfile.py:26]:
```suggestion
        # TODO: self.requires("cpython/x.y/z") or self.requires("pypy/x.y/z")
```

Make easier to be found in the future

File: recipes/pybind11/all/conanfile.py [hunk 5]
```diff
@@ -16,17 +16,25 @@ class PyBind11Conan(ConanFile):
...
 
+    _cmake = None
+
     def source(self):
         tools.get(**self.conan_data["sources"][self.version])
         os.rename("{}-{}".format(self.name, self.version), self._source_subfolder)
 
+    def requirements(self):
+        # TODO: self.requires("cpython/x.y/z") or self.requires("pypy/x.y/z")
+        self.output.info("This recipe requires a python SDK which is not available (yet) on CCI")
```
Comment (madebr) [inline recipes/pybind11/all/conanfile.py:27]: @danimtb  LIke this?
```suggestion
        pass
```

Comment (danimtb) [reply recipes/pybind11/all/conanfile.py:27]: I'd rather remove the requirements() method completely. We we have something like python I am sure we will revisit recipes that require python somehow

Comment (madebr) [reply recipes/pybind11/all/conanfile.py]: Ok. Removed it.

**Comment by gocarlos** [comment]
already two people taking the effort to upgrade the pybind11 version as this one still not on master...
:(

**Comment by uilianries** [comment]
@madebr we have conflict with your PR. Could you please solve it?

**Comment by madebr** [comment]
@uilianries 
> @madebr we have conflict with your PR. Could you please solve it?

**DONE**

**Comment by uilianries** [comment]
@madebr thanks a lot!

**Pull request merged**